pkg/common/ingress/controller/controller.go

```diff
@@ -136,7 +137,8 @@ const IngressClassKey = "kubernetes.io/ingress.class"
 // IsValidClass ...
 func (ic *GenericController) IsValidClass(ing *extensions.Ingress) bool {
 	ann, found := ing.Annotations[IngressClassKey]
-	return !found || ann == ic.cfg.IngressClass
+
+	return (ic.cfg.IgnoreIngressWithoutClass && found && ann == ic.cfg.IngressClass) || (!ic.cfg.IgnoreIngressWithoutClass && (!found || ann == ic.cfg.IngressClass))
```

Comment:
Change to a simpler approach, eg (didn't test)

```go
if !found {
  return !ic.cfg.IgnoreIngressWithoutClass
}
return ann == ic.cfg.IngressClass
```

Reply:
made the changes, please verify